pyvista/plotting/_property.py

```diff
@@ -0,0 +1,1144 @@
+
+        Visualize metallic at ``0.5``.
+
+        >>> prop.metallic = 0.5
+        >>> prop.plot()
+
+        Visualize metallic at ``1.0``.
+
+        >>> prop.metallic = 1.0
+        >>> prop.plot()
+
+        """
+        _check_supports_pbr()
+        return self.GetMetallic()
+
+    @metallic.setter
+    def metallic(self, value: float):
+        _check_supports_pbr()
+        self.SetMetallic(value)
+
+    @property
+    def roughness(self) -> float:
+        """Return or set roughness.
+
+        This requires that the interpolation be set to ``'Physically based
+        rendering'``
+
+        Examples
+        --------
+        Set roughness to 0.1
+
+        >>> import pyvista as pv
+        >>> prop = pv.Property()
+        >>> prop.interpolation = 'pbr'  # requires physically based rendering
+        >>> prop.metallic = 0.5  # helps to visualize metallic
+        >>> prop.roughness = 0.1
+        >>> prop.roughness
+        0.1
+
+        Visualize default roughness with metallic of ``0.5``.
+
+        >>> prop.roughness = 0.5
+        >>> prop.plot()
+
+        Visualize roughness at ``0.0`` with metallic of ``0.5``.
+
+        >>> prop.roughness = 0.0
+        >>> prop.plot()
+
+        Visualize roughness at ``1.0`` with metallic of ``0.5``.
+
+        >>> prop.roughness = 1.0
+        >>> prop.plot()
+
+        """
+        _check_supports_pbr()
+        return self.GetRoughness()
+
+    @roughness.setter
+    def roughness(self, value: bool):
+        _check_supports_pbr()
+        self.SetRoughness(value)
+
+    @property
+    def interpolation(self) -> str:
+        """Return or set the method of shading.
+
+        One of the following options.
+
+        * ``'Physically based rendering'`` - Physically based rendering.
+        * ``'pbr'`` - Alias for Physically based rendering.
+        * ``'Phong'`` - Phong shading.
+        * ``'Gouraud'`` - Gouraud shading.
+        * ``'Flat'`` - Flat Shading.
+
+        This is really great.
+
+        Examples
+        --------
+        Set interpolation to physically based rendering.
+
+        >>> import pyvista as pv
+        >>> prop = pv.Property()
+        >>> prop.interpolation = 'pbr'
+        >>> prop.interpolation
+        'Physically based rendering'
+
+        Visualize default flat shading.
+
+        >>> prop.interpolation = 'Flat'
+        >>> prop.plot()
+
+        Visualize gouraud shading.
+
+        >>> prop.interpolation = 'Gouraud'
+        >>> prop.plot()
+
+        Visualize phong shading.
+
+        >>> prop.interpolation = 'Phong'
+        >>> prop.plot()
+
+        Visualize physically based rendering.
+
+        >>> prop.interpolation = 'Physically based rendering'
+        >>> prop.plot()
+
+        """
+        return self.GetInterpolationAsString()
+
+    @interpolation.setter
+    def interpolation(self, value: str):
+        if not isinstance(value, str):
+            raise TypeError(f'`interpolation` must be a string, not {type(value)}')
```

Comment:
This seems to clash with "`None` means flat" now.

Reply:
Completely removed the mention of `None` in f76f8f05fd87a400f6b5f03df9f02a7af3df2df6.